### DISCUSSION TITLE
How to handle vtkPartitionedDataSetCollections in Trame?

### DISCUSSION THREAD

**Discussion by dougbrennan** [Kitware/trame#508] [General] [answered]
Hello!!!
I'm working with vtkPartitionedDataSetCollections and I find I cannot display them in Trame with local mode enabled on my VtkRemoteLocalView.
Works just fine for remote - but I need local..... 
The pipeline incudes a vtkGeometryFilter to extract the free faces from solids in th vtkPartitionedDataSetCollection and then I use vtkCompositePolyDataMapper2. I also tried vtkCompositePolyDataMapper with the same result. 
There are no errors reported, but the display is empty.
Is this an expected limitation?
Is there a known approach that works?

**Comment by jourdain** [comment, marked as answer]
You are hitting vtk.js limitation but VTK/WASM aim to fix those. 
While not ready for prime time (we could be missing serializer), you can try to use [trame-vtklocal](https://github.com/Kitware/trame-vtklocal) instead of trame-vtk. But then you will just have a local view...

**Reply by dougbrennan** [reply]
I worked around it by merging all of the polys from the collection into a single poly. Which seems like it should have been easier than it was - I basically had to loop over the collection and add each each poly to an append filter. I couldn't find a filter that did it directly. I know Paraview has one but I'm working in VTK at the moment.

I really want to keep my options open on local/remote for now. I suspect I'll have use cases for both.

Thanks for the suggestion and confirming the limitation!!!
